### ISSUE TITLE
How to extract more info from django request?

### ISSUE THREAD

**Issue by teeberg** [getsentry/sentry-python#95]
The user data that's logged by default (with sending of PII enabled) looks something like
```
{'username': teeberg', 'ip_address': '127.0.0.1', 'email': teeberg@example.com'}
```
I'm trying to add some fields to this, but there doesn't seem to be an official documented way to do this.

Is there a cleaner way than monkey-patching `sentry_sdk.integrations.django._set_user_info`?

Maybe the request could be sent along as another hint to `before_send` handlers?

**Comment by untitaker** [comment]
Hi, does this help? https://docs.sentry.io/learn/context/?platform=python#capturing-the-user

Which fields do you want to add?

**Comment by teeberg** [comment]
That doc only explains how to set the user to a fixed string, but how do I get it from the current request? How do I access the current request in a `with configure_scope()` statement? I'd like to add various flags, such as:
- replacing the IP with the actual IP, since sentry doesn't infer it correctly
- the access level of the user
- If somebody uses "View as" functionality, who the real user behind the request is, additionally to the assumed "actual" user
- Which organization they belong to

**Comment by teeberg** [comment]
I just realize that I was probably too tunnel-visioned. I guess I can just add a custom middleware or something that sets the user the second a request comes in... I was looking for a way to do it with the SDK for some reason. If that's the recommended way, please close :)

**Comment by untitaker** [comment]
Yes, that's the way to do it, you always configure the scope for each request :) One thing to note:

>replacing the IP with the actual IP, since sentry doesn't infer it correctly

* you currently can't set the IP manually, the integration will later override it again. I opened #96 for this.
* the fact that the SDK doesn't find the correct IP sounds like worth investigating. Do you know what causes this? Did this work in the old SDK?

**Comment by untitaker** [comment]
@teeberg just re-read your message, what built-in hooks are you referring to?

**Comment by teeberg** [comment]
> what built-in hooks are you referring to?

With the old python raven library, I subclassed `DjangoClient` and overrode `get_user_info` and `get_data_from_request`. Don't remember if that was documented somewhere or how I get there, but I suspect the middleware didn't work for some reason. Looks like it got the user info directly from the request without providing a way to hook into and extend it. Anyway, that's what I did back then, I think I was still assuming that a middleware wouldn't work 🙄

> the fact that the SDK doesn't find the correct IP sounds like worth investigating. Do you know what causes this? Did this work in the old SDK?

Sorry, this was actually me speaking from memory about what happened with the old library. I saw code in the new library to deal with reverse proxies, so maybe that'll be fixed now. I'll open a new issue if I still see it inferring the wrong IP.

On a somewhat related note, so far I'm really liking the efforts to make the different integrations agree on a common format/API! And also the API seems pretty well thought out, nice work on that 👍🏻
